pkg/geo/geomfn/node.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/twpayne/go-geom"
+	"github.com/twpayne/go-geom/xy"
+)
+
+// extractUniqueEndpoints extracts the endpoints from geometry and removes duplicates
+func extractUniqueEndpoints(g geo.Geometry) (geo.Geometry, error) {
+	endpoints, err := extractEndpoints(g)
+	if err != nil {
+		return geo.Geometry{}, fmt.Errorf("error extracting endpoints: %v", err)
+	}
+
+	output, err := UnaryUnion(endpoints)
+	if err != nil {
+		return geo.Geometry{}, fmt.Errorf("error removing duplicates: %v", err)
+	}
+	return output, nil
+}
+
+// extractEndpoints extracts the endpoints from geometry provided and returns them as a multipoint geometry
```

Comment:
nit: `.` at end of sentence.
spell multipoint as MultiPoint

Reply:
done